src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotPassTypesByReference.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotPassTypesByReference : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1045";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.DoNotPassTypesByReferenceTitle), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.DoNotPassTypesByReferenceMessage), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.DoNotPassTypesByReferenceDescription), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(
+            RuleId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.Design,
+            RuleLevel.IdeSuggestion,
```

Comment:
What shall be the default level?

Reply:
From my perspective, Disabled :-)  It's nice that we're aiming to port all of the old FxCop rules so that folks that still want them can use them.  But this rule in particular feels very outdated, and I expect it will drive the wrong behaviors.  For example, my understanding is this would warn any time someone tries to write a method that uses the Try pattern.

Reply:
I agree - most of the ported FxCop rules should be disabled by default, so users can turn them on explicitly if they desire. Only a handful of these would be on by default suggestions, if they are indeed helpful code improvements, such that analyzer has no noise.

Reply:
NOTE: None of the ported FxCop rules should be enabled warnings by default - I am going to write a special analyzer for this repo which helps enforce that only design team approved analyzers are marked as enabled warnings.

Reply:
@stephentoub This rule only looks at `ref` not `out` parameters so the Try pattern is not impacted here. This is the rule `AvoidOutParameters` that takes care of it.

Reply:
Ok.  Then lazy initialization methods :)